src/main/java/io/github/thebusybiscuit/slimefun4/api/accessibility/AccessManager.java

```diff
@@ -0,0 +1,166 @@
+     *
+     * @param clazz The instance of the class
+     * @param <T>   Generic type of the class
+     * @return Returns a {@link Optional} of the AccessData object backing the provided class.
+     * @see #getOrRegisterAccessData(Class, AccessData)
+     */
+    @Nonnull
+    <T> Optional<AccessData<T>> getAccessData(@Nonnull Class<T> clazz);
+
+    /**
+     * Get the {@link AccessData} object backing a certain class, or register a default if
+     * {@link #getAccessData(Class)} is not present. Mutations to the returned
+     * object will be reflected by this manager.
+     *
+     * @param clazz The instance of the class
+     * @param <T>   Generic type of the class
+     * @return Returns the AccessData object backing the provided class
+     */
+    @Nonnull
+    <T> AccessData<T> getOrRegisterAccessData(@Nonnull Class<T> clazz,
+                                              @Nonnull final AccessData<T> def);
+
+    /**
+     * Check whether a {@link Player} has a given {@link AccessLevel} or higher.
+     *
+     * @param player      The instance of the player
+     * @param accessLevel The instance of the AccessLevel
+     * @return Returns true if the player has the provided access level or higher.
+     */
+    default boolean hasLevel(@Nonnull Player player, @Nonnull AccessLevel accessLevel) {
+        return getAccessLevel(player).compare(accessLevel) >= 0;
+    }
+
+    /**
+     * Check whether a player has a given {@link AccessLevel} or higher.
+     *
+     * @param player      The unique id of the player
+     * @param accessLevel The instance of the AccessLevel
+     * @return Returns true if the player has the provided access level or higher.
+     */
+    default boolean hasLevel(@Nonnull UUID player, @Nonnull AccessLevel accessLevel) {
+        return getAccessLevel(player).compare(accessLevel) >= 0;
+    }
+
+    /**
+     * Check whether the manager has data for a given object. This method is useful in
+     * understanding whether a player is on the access list when {@link #getAccessLevel(Player)}
+     * provides the default result.
+     *
+     * @param object The instance of the object.
+     * @return Returns whether the manager has data for a given object.
+     */
+    boolean hasDataFor(@Nonnull Object object);
+
+    /**
+     * @return Returns whether this access manager has no data.
+     */
+    boolean isEmpty();
+
+    /**
+     * Reset this manager to an empty state.
+     */
+    void reset();
+
+    /**
+     * Purge all currently held data and read from the {@link ItemStack}
+     *
+     * @param itemStack The instance of the {@link ItemStack}
+     */
+    void load(@Nonnull NamespacedKey namespace, @Nonnull ItemStack itemStack);
+
+    /**
+     * Purge all currently held data and read from the {@link Block}
+     *
+     * @param block The instance of the {@link Block}
+     */
+    void load(@Nonnull NamespacedKey namespace, @Nonnull Block block);
```

Comment:
Not sure if it is a typo but if it really "purges" then the term "reload" would be more appropriate.